src/recipes/Viewer.tsx

```diff
@@ -0,0 +1,279 @@
+import { PureComponent, default as React } from "react";
+import { StyleSheet, View } from "react-native";
+import {
+  CircularButton,
+  ConfirmationPopUp,
+  InfoCard,
+  Text
+} from "../components";
+import { colors } from "../theme";
+import Icon from "@anarock/pebble/native/Icon";
+import Select from "../components/Select";
+import Button from "../components/Button";
+import nI from "name-initials";
+import { ViewerProps, ViewerState } from "./typings/Viewer";
+
+const styles = StyleSheet.create({
+  circButton: {
+    marginBottom: 0,
+    marginRight: 15
+  },
+  container: {
+    flexDirection: "row",
+    alignItems: "center",
+    paddingVertical: 20
+  },
+  rowLeft: { flex: 1, flexDirection: "row", alignItems: "center" }
+});
+
+const viewerInfoCard = StyleSheet.create({
+  container: {
+    marginBottom: 0
+  },
+  topSection: {
+    paddingHorizontal: 0,
+    paddingTop: 0,
+    paddingBottom: 0
+  },
+  heading: {
+    marginBottom: 5
+  }
+});
+
+export default class extends PureComponent<ViewerProps, ViewerState> {
+  selectRef: React.RefObject<Select> = React.createRef();
+
+  state = {
+    showUnfollowConfirmation: false,
+    selectedAgentId: null,
+    showTransferConfirmation: false,
+    showTransferAndFollowConfiguration: false
+  };
+
+  private openTransferConfirmationModal = () =>
```

Comment:
Verbose, instead of opens and closes. You could just have 1 for each.
- toggleTransferConfirmationModal
- toggleTransferAndFollowConfigurationModal
- UnfollowConfirmationModal 

`toggleModalType = () => this.setState( {modalState} => updatedModalState: !modalState);`